### ISSUE TITLE
'inference_input_size' hyperparameter (recurrent methods) conflict with 'cross_validation'

### ISSUE THREAD

**Issue by PotosnakW** [Nixtla/neuralforecast#531]
**Describe the bug**
The 'inference_input_size' hyperparameter (recurrent methods) causes an issue with 'cross_validation' when generating the final prediction .csv file. The prediction output is in the shape of the windows [batch, input_size, horizon] (generated from the _create_windows function in _base_recurrent.py). However, the fcsts variable (in the cross_validation function in core.py) requires a larger shape pertaining to the training set. A screenshot of the error is provided below.

**To Reproduce**
1. Run the example code below
2. See error

**Expected behavior**
Cross validation code generates final prediction .csv file without errors.

**Screenshots**
Error image below: 
<img width="1198" alt="example_issue_cross_val" src="https://user-images.githubusercontent.com/71658863/233179365-b57b3277-c47d-44cf-96c7-f1e074c8a8ca.png">

**Desktop (please complete the following information):**
- OS: Springdale Open Enterprise Linux (--> based on Red Hat Enterprise Linux)
- Browser: N/A
- Version: 8.7 Moderna

**Additional context**
**Example Code Below**

import pandas as pd
from datasetsforecast.long_horizon import LongHorizon

from neuralforecast.auto import AutoRNN, AutoLSTM, AutoTCN
from neuralforecast.core import NeuralForecast
from neuralforecast.losses.pytorch import MAE

import logging
logging.getLogger("pytorch_lightning").setLevel(logging.WARNING)

Y_df, _, _ = LongHorizon.load(directory='./', group='ETTm2')
Y_df['ds'] = pd.to_datetime(Y_df['ds'])

n_time = len(Y_df.ds.unique())
val_size = int(.2 * n_time)
test_size = int(.2 * n_time)
horizon = 96
freq='15min'
    
rnn_config = {'input_size': 2*horizon,
             'inference_input_size': 2*horizon,
            'max_steps':5,
            'val_check_steps': 2,
            'random_seed': 1, 
            }
            
fcst = NeuralForecast(freq=freq,
                        models=[AutoRNN(h=horizon,
                                    config=rnn_config,
                                    loss=MAE()),
                                    ],)

fcst_df = fcst.cross_validation(df=Y_df, 
                                val_size=val_size,
                                test_size=test_size, 
                                n_windows=None)
